dependencies.yaml

```diff
@@ -0,0 +1,164 @@
+          - cxx-compiler
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              arch: x86_64
+            packages:
+              - gcc_linux-64=9.*
+              - sysroot_linux-64=2.17
+          - matrix:
+              arch: aarch64
+            packages:
+              - gcc_linux-aarch64=9.*
+              - sysroot_linux-aarch64=2.17
+  checks:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - pre-commit
+  cudatoolkit:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              cuda: "11.8"
+              arch: x86_64
+            packages:
+              - cudatoolkit=11.8
+              - libcufile=1.4.0.31
+              - libcufile-dev=1.4.0.31
+          - matrix:
+              cuda: "11.8"
+              arch: aarch64
+            packages:
+              - cudatoolkit=11.8
+          - matrix:
+              cuda: "11.5"
+            packages:
+              - cudatoolkit=11.5
+              - libcufile>=1.1.0.37,<=1.1.1.25
+              - libcufile-dev>=1.1.0.37,<=1.1.1.25
+          - matrix:
+              cuda: "11.4"
+            packages:
+              - cudatoolkit=11.4
+              - &libcufile_114 libcufile>=1.0.0.82,<=1.0.2.10
+              - &libcufile_dev114 libcufile-dev>=1.0.0.82,<=1.0.2.10
+          - matrix:
+              cuda: "11.2"
+            packages:
+              - cudatoolkit=11.2
+              # The NVIDIA channel doesn't publish pkgs older than 11.4 for these libs,
+              # so 11.2 uses 11.4 packages (the oldest available).
+              - *libcufile_114
+              - *libcufile_dev114
+  docs:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - pydata-sphinx-theme
+          - sphinx<6
+      - output_types: conda
+        packages:
+          - doxygen=1.8.20 # pre-commit hook needs a specific version.
+  py_version:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              py: "3.8"
+            packages:
+              - python=3.8
+          - matrix:
+              py: "3.9"
+            packages:
+              - python=3.9
+          - matrix:
+              py: "3.10"
+            packages:
+              - python=3.10
+          - matrix:
+            packages:
+              - python>=3.8,<3.11
+  run:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - numpy
+      - output_types: conda
+        packages:
+          - cupy>=7.8.0,<12.0.0a0
+    specific:
+      - output_types: requirements
+        matrices:
+          - matrix:
+              arch: x86_64
+            packages:
+              - cupy-cuda115>=9.5.0,<12.0.0a0 # TODO: This might change to cupy-cuda11x?
+          - matrix:
+              arch: aarch64
+            packages:
+              - cupy-cuda11x -f https://pip.cupy.dev/aarch64 # TODO: Verify that this works.
```

Comment:
@AjayThorve, can you make this section match the `run` sections the recipes' `meta.yaml` files?

You can omit packages like `python` and `cudatoolkit` since they're covered in other dependency lists in this file.

Reply:
updated, let me know if that looks good now @ajschmidt8

Reply:
thanks! pushed a few small tweaks

Reply:
thanks!